src/structures/Application.ts

```diff
@@ -0,0 +1,59 @@
+import { APIApplication, ApplicationFlags, Snowflake } from 'discord-api-types';
+import { Client } from '../clients';
+import { deconstruct } from '../utils/Snowflake';
+import { Team } from './team/Team';
+import { User } from './User';
+
+export class Application {
+	public id?: Snowflake;
```

Comment:
are you sure that id is nullable ?

Reply:
Not directly on APIApplication, but Invite has a target_application property that is a partial APIApplication, so it can be nullable.

Reply:
even the id?

Reply:
cf. https://discord.com/developers/docs/topics/oauth2#application

Reply:
It says "partial application object", we don't really know what is always sent : https://discord.com/developers/docs/resources/invite#invite-object-invite-structure

Reply:
yes, but logically, the id cannot be absent.